packages/odf/components/topology/TopBar/SearchComponent.tsx

```diff
@@ -0,0 +1,124 @@
+  const [searchInput, setSearchInput] = React.useState('');
+
+  const { visualizationLevel, nodes, deployments } =
+    React.useContext(TopologyDataContext);
+
+  const { activeItemsUID, setActiveItemsUID } = React.useContext(
+    TopologySearchContext
+  );
+
+  const controller = useVisualizationController();
+
+  const performSearch = React.useCallback(
+    (input) => {
+      setSearchInput(input);
+      let resourcesInScope: K8sResourceCommon[] = nodes;
+      if (visualizationLevel === TopologyViewLevel.DEPLOYMENTS) {
+        const visibleElements = controller.getElements();
+        const visibleIDs = visibleElements.map((element) => element.getId());
+        const filteredDeployments = deployments.filter((deployment) =>
+          visibleIDs.includes(deployment.metadata.uid)
+        );
+        resourcesInScope = filteredDeployments;
+      }
+      // Process the search now
+      const filteredItems = resourcesInScope
+        .filter((item) => {
+          const itemName = getName(item);
+          if (fuzzy(searchInput, itemName)) {
+            return true;
+          }
+          return false;
+        })
+        .map(getUID);
+      setActiveItemsUID(filteredItems);
+      return filteredItems;
+    },
+    [nodes, visualizationLevel, setActiveItemsUID, deployments, searchInput]
+  );
+
+  return { performSearch, activeItemsUID };
+};
+
+const SearchBar: React.FC = () => {
+  const [userInput, setUserInput] = React.useState('');
+  const [currentResult, setCurrentResult] = React.useState(0);
+  const controller = useVisualizationController();
+
+  const { performSearch, activeItemsUID } = useTopologySearch();
+
+  const { setActiveItemsUID, setActiveItem } = React.useContext(
+    TopologySearchContext
+  );
+
+  const resultsCount = activeItemsUID.length;
+
+  const { t } = useCustomTranslation();
+
+  const onChange = (event, input: string) => {
+    setUserInput(input);
+  };
+
+  const onClear = () => {
+    setUserInput('');
+    setCurrentResult(0);
+    setActiveItemsUID([]);
+    setActiveItem('');
+  };
+
+  const onSearch = () => {
+    const items = performSearch(userInput);
+    setActiveItem(items[0]);
+  };
+
+  const onNext = () => {
+    const newCurrentResult = currentResult + 1;
+    setCurrentResult(
+      newCurrentResult > resultsCount ? resultsCount : newCurrentResult
+    );
+    setActiveItem(activeItemsUID[newCurrentResult]);
+  };
+
+  const onPrevious = () => {
+    const newCurrentResult = currentResult - 1;
+    setCurrentResult(newCurrentResult >= 0 ? newCurrentResult : 1);
+    setActiveItem(activeItemsUID[newCurrentResult]);
+  };
+
+  React.useEffect(() => {
+    controller.addEventListener('STEP_INTO_EVENT', onClear);
+  }, []);
+
+  return (
+    <SearchInput
+      isNextNavigationButtonDisabled={resultsCount === currentResult + 1}
+      isPreviousNavigationButtonDisabled={currentResult === 0}
+      value={userInput}
+      placeholder={t('Search..')}
```

Comment:
nits: super minor comment, I have seen three dots in console for a placeholder...
```suggestion
      placeholder={t('Search...')}
```